### PR TITLE
Fix routing

### PR DESCRIPTION
**Pull request by drewdrewthis**
- Updates code so that isolationMode works without refresh
- Also adds nicer UX that updates browser address bar when different lists are selected

NOTE: This branch is off of this branch: https://github.com/policygenius/react-styleguidist/pull/9

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -109,6 +109,7 @@
     "preact-compat": "^3.16.0",
     "prettier": "^1.3.1",
     "react": "^15.5.4",
+    "react-addons-perf": "^15.4.2",
```
Comment (patrickkim) [inline package.json]: did you mean to include this?

File: src/rsg-components/StyleGuide/StyleGuide.js [hunk 2]
```diff
@@ -51,23 +52,34 @@ export default class StyleGuide extends Component {
...
 
-		return list;
+	getSectionsByListType(selectedListType, sections) {
+		if (sections.length === 1) {
+			return sections;
+		}
+
+		const list = sections.find(section => section.name === selectedListType);
+
+		if (list) {
```
Comment (patrickkim) [inline src/rsg-components/StyleGuide/StyleGuide.js]: You can replace this conditional with `return list.sections || []`

Comment (drewdrewthis) [inline src/rsg-components/StyleGuide/StyleGuide.js]: @patrickkim No.. if `list` is undefined, `list.sections` will throw an error.

File: package.json (hunk 1 above)
Comment (drewdrewthis) [inline package.json]: Nope. Removing.

**Pull request merged**